src/sentry/data_export/tasks.py

```diff
@@ -308,7 +308,8 @@ def merge_export_blobs(data_export_id, **kwargs):
 
         # adapted from `putfile` in  `src/sentry/models/file.py`
         try:
-            with transaction.atomic():
+            assert router.db_for_write(File) == router.db_for_write(FileBlobIndex)
+            with transaction.atomic(using=router.db_for_write(File)):
```

Comment:
Probably same comment here as earlier, we might need a transaction on both dbs since `ExportedDataBlob` is on the old db